### PR TITLE
test: integration testing of functionality

### LINKED ISSUE #121: Intergration testing of functionality
Test each individual command functionality and the chaining of all available commands  the

-     Read SBOMs into the cache
-     Merge SBOMs
-     List SBOMs in the cache
-     Output SBOMs from the cache in a sbom format of your choice
-     Aliasing

### PR DESCRIPTION
**Pull request by ashearin**
## Description

Adding e2e tests using testscript package.

There are commented out tests/lines in different files for failures that weren't expected and were not a quick fix. I attempted to record the failure reason, so if you have more context on a failure or know how to fix the issue please lmk. Otherwise a follow on issue will be created for each. 

Also made an update to the way the root cmd is called, as it was needed for this test package.

These tests can be run as part of the general 'go test' cmd, but I added tasks in the task file and Makefile to run only the e2e and unit tests. 

Fixes #121

## References (More info on testscript)
- [blog post](https://bitfieldconsulting.com/posts/tag/testscript)
- [slides](https://fosdem.org/2024/events/attachments/fosdem-2024-1802-testing-go-command-line-programs-with-go-internal-testscript-/slides/22802/Testing_Go_programs_with_go-internal-testscript_wT7GqqA.pdf)
- [golang docs](https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript)
- [example repo](https://github.com/datacharmer/wordcount/tree/main)

### CODE REVIEW

**Review by lmphil — APPROVED** [review]

File: internal/e2e/e2eutil/setup_cache.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		}
+
+		sbomData, err := os.ReadFile(filepath.Join(testDataDir, name))
+		script.Check(err)
+
+		doc, err := backend.AddDocument(sbomData)
+		script.Check(err)
+
+		// POTENTIAL BUG: Unable to access this bom via ID without an alias, Works when run manually in cli
+		if strings.Contains(name, "spdx") {
```
Comment (ashearin) [inline internal/e2e/e2eutil/setup_cache.go:82]: Workaround related to spdx id issue in test env. to be resolved in another issue.

**Comment by jhoward-lm** [comment]
For the revive `confusing-naming` linter issue, I think it's ok to add the following to `.golangci.yml` under `linter-settings:revive:rules`:

```yaml
      - name: confusing-naming
        exclude: [TEST]
```

File: internal/e2e/export/export.txtar [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+stdout '.*("documentNamespace": "https://spdx.org/spdxdocs/").*'
+
+# export -o spdx sbom
+! exists pikachu.json
+exec bomctl export --cache-dir $WORK -o pikachu.json -f spdx spdx-4
+! stdout .
+stderr '^(INFO  export: Exporting document sbomID=https://anchore.com/syft/file/bomctl_0.3.0_linux_amd64.tar.gz-1b838d44-9d3c-47d0-9f7f-846397e701fa#DOCUMENT)$'
+exists pikachu.json
+# compare_docs $WORK pikachu.json test-linked.spdx.json
+# Fails with MetaData values differing:
```
[the remark this answers is not in the document]
Comment (ashearin) [reply internal/e2e/export/export.txtar:65]: to be resolved in https://github.com/bomctl/bomctl/issues/178

File: internal/e2e/e2eutil/setup_cache.go (hunk 1 above)
Comment (ashearin) [reply internal/e2e/e2eutil/setup_cache.go]: This issue has been resolved, see https://github.com/bomctl/bomctl/pull/169/commits/8750098dc75dcf74cd6f1aa9dd394a090e6bfd1c

**Review by jhoward-lm — APPROVED** [review]
Looks good, thanks!

**Pull request merged**